src/http/server.cr

```diff
@@ -296,8 +296,20 @@ class HTTP::Server
 
   # Adds a `Socket::Server` *socket* to this server.
   def bind(socket : Socket::Server) : Nil
-    raise "Can't add socket to running server" if listening?
-    raise "Can't add socket to closed server" if closed?
+    bind(socket) { }
+  end
+
+  # Helper method which yields to block, allowing to close a newly created socket if
+  # it can't be bound to the HTTP server. This avoids re-raising the error.
```

Comment:
Shouldn't we use `rescue` in callers for the cleanup, rather than a custom method?

Reply:
Yes, using the public API it would be the proper way to rescue the exception, close the socket and then re-raise. I used this solution to avoid the additional cost for that. It's not a huge gain though, considering this exception should be pretty rare. I can change it to `rescue` + re-raise if that's not welcome.

Reply:
I'd prefer the cleaner code here, this code is unlikely to ever be on the hot path.